Add first draft for descriptive analysis (EDA #1)

valbo451 opened the pull request:
I added the "descriptive-analysis.py" file to our back-end with the following functions:

- **import_csv**: Read in the dataset and return a DataFrame
- **calculate_basic_statistics** : Returns dict with statistical attributes (for continuous column) or binary distribution (for binary column)
- **save_variable_distribution**; **save_distribution_plot**; **show_plot**: Three functions to count variable expressions and visualize them in a simple bar plot
- **create_conditional_dataset**: Function to create a conditional subset of our dataset (e.g. return subset for for patients over 60)

Additionally, a dictionary **MEANING_BINARY_COLUMNS** that maps the binary digits of our binary columns to concrete meanings.

HokageM commented on src/heartpredict/backend/descriptive-analysis.py:
```diff
@@ -0,0 +1,131 @@
+}
+
+
+def import_csv(csv_path: str = "data/heart_failure_clinical_records.csv"):
```
this can be already found in the io module

HokageM commented on src/heartpredict/backend/descriptive-analysis.py:
```diff
@@ -0,0 +1,131 @@
+    Determine basic statistics for a given column
+
+    Input:
+    DataFrame of (sub)dataset
```
```suggestion
    df: DataFrame of (sub)dataset
```

HokageM commented on src/heartpredict/backend/descriptive-analysis.py:
```diff
@@ -0,0 +1,131 @@
+
+    Input:
+    DataFrame of (sub)dataset
+    Column to be analyzed
```
```suggestion
    col: Column to be analyzed
```

HokageM commented:
Use this type of function documentation:

```
"""
Describtion of function.
Args:
     x: describtion of var x
     y: ...

Returns:
     describtion of return
"""

valbo451 commented:
With the last three commits, I've addressed the review:

- Correct function docstrings
- Remove import_csv function

**And did this:**
- Remove unused libraries
- Add global docstring
- Rename file from "descriptive-analysis" to "descriptive" to enable import as a module
